[BE] Simplify tests, elaborate testnames in test_optim.py

janeyx99 opened the pull request:
- Deletes unused kwargs
- Make test names more descriptive to remove need of comments. Overall it's better to codify over comment
- Added a test for duplicate params across groups
- Greatly simplified test_empty_grad to discover that the crux of the bug was NOT its emptiness, but rather with multi-dim emptiness.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #101103
* #101100
* __->__ #101004

janeyx99 committed:
[BE] Simplify tests, elaborate testnames in test_optim.py

[ghstack-poisoned]

janeyx99 committed:
Update on "[BE] Simplify tests, elaborate testnames in test_optim.py"

- Deletes unused kwargs
- Make test names more descriptive to remove need of comments. Overall it's better to codify over comment
- Added a test for duplicate params across groups
- Greatly simplified test_empty_grad to discover that the crux of the bug was NOT its emptiness, but rather with multi-dim emptiness.




[ghstack-poisoned]

albanD commented on test/test_optim.py:
```diff
@@ -1500,9 +1498,6 @@ def test_asgd(self):
                     [params], maximize=False, weight_decay=0.9, foreach=foreach
                 )
             )
-            self._test_complex_optimizer(
```
This is a duplicate?

albanD commented on test/test_optim.py:
```diff
@@ -1568,41 +1563,93 @@ def closure():
         with warnings.catch_warnings(record=True) as w:
             warnings.simplefilter("always")
-            optim.SGD([param, param], lr=0.1)
+            optim.SGD([param, param], lr=0.01)
```
nit: `self.assertWarnsRegex()`

janeyx99 replied:
:o til

albanD reviewed (commented):
Sounds pretty good!

janeyx99 committed:
Update on "[BE] Simplify tests, elaborate testnames in test_optim.py"

- Deletes unused kwargs
- Make test names more descriptive to remove need of comments. Overall it's better to codify over comment
- Added a test for duplicate params across groups
- Greatly simplified test_empty_grad to discover that the crux of the bug was NOT its emptiness, but rather with multi-dim emptiness.




[ghstack-poisoned]

janeyx99 committed:
Update on "[BE] Simplify tests, elaborate testnames in test_optim.py"

- Deletes unused kwargs
- Make test names more descriptive to remove need of comments. Overall it's better to codify over comment
- Added a test for duplicate params across groups
- Greatly simplified test_empty_grad to discover that the crux of the bug was NOT its emptiness, but rather with multi-dim emptiness.




[ghstack-poisoned]

janeyx99 committed:
Update on "[BE] Simplify tests, elaborate testnames in test_optim.py"

- Deletes unused kwargs
- Make test names more descriptive to remove need of comments. Overall it's better to codify over comment
- Added a test for duplicate params across groups
- Greatly simplified test_empty_grad to discover that the crux of the bug was NOT its emptiness, but rather with multi-dim emptiness.




[ghstack-poisoned]

albanD reviewed (approved):
Ok!

janeyx99 committed:
Update on "[BE] Simplify tests, elaborate testnames in test_optim.py"

- Deletes unused kwargs
- Make test names more descriptive to remove need of comments. Overall it's better to codify over comment
- Added a test for duplicate params across groups
- Greatly simplified test_empty_grad to discover that the crux of the bug was NOT its emptiness, but rather with multi-dim emptiness.




[ghstack-poisoned]

janeyx99 commented:
@pytorchbot merge
